src/App.tsx

```diff
@@ -1,95 +1,68 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
+const itemsPerPage: string[] = ['3', '5', '10', '20'];
+
+const total = items.length;
+
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const lastItemOnPage = Math.min(currentPage * perPage, total);
+  const firstItemOnPage = ((currentPage - 1) * perPage) + 1;
+  const filteredItems = items.slice((currentPage - 1)
+  * perPage, currentPage * perPage);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${firstItemOnPage} - ${lastItemOnPage} of ${total})`}
       </p>
-
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            onChange={(event) => {
+              setPerPage(+event.target.value);
+              setCurrentPage(1);
+            }}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here